service.go

```diff
@@ -184,3 +186,43 @@ func prepArgusListenerClientConfig(cfg *ListenerConfig, watches ...Watch) {
 		}
 	})
 }
```

Comment:
brought all this over from caduceus (https://github.com/xmidt-org/caduceus/blob/denopink/feat/rewrite/internal/anclaHelper/anclaHelper.go) 

stopWatches, err := svc.StartListener(in.Listener, setLoggerInContext(), in.Sink) remains in caduceus draft pr